sdk/samples/iot/templates/iothub/test-resources-post.ps1

```diff
@@ -17,6 +17,29 @@ param (
     $RemainingArguments
 )
 
+function waitForActiveHub
+{
+    $retryCount = 0
+    do
+    {
+      $retryCount++
+      Write-Host "AzIotHub is not yet active so sleeping for $retryCount seconds."
+      Start-Sleep -Seconds $retryCount
```

Comment:
Nit: Shouldn't we retry with exponential back-off? (Here and elsewhere where we retry.)

`Start-Sleep -Seconds [int][Math]::Pow(2, $retryCount)`

From what I've seen with my subscriptions, provisioning a Hub or DPS can take minutes in some cases.